src/controllers/DrawingModeController.js

```diff
@@ -149,11 +120,61 @@ class DrawingModeController extends AnnotationModeController {
                 this.saveThread(this.currentThread);
             }
 
+            this.currentThread = undefined;
             this.toggleMode();
         });
 
-        this.pushElementHandler(this.undoButtonEl, 'click', this.currentThread.undo);
-        this.pushElementHandler(this.redoButtonEl, 'click', this.currentThread.redo);
+        this.pushElementHandler(this.undoButtonEl, 'click', () => {
+            if (this.currentThread) {
+                this.currentThread.undo();
+            }
+        });
+
+        this.pushElementHandler(this.redoButtonEl, 'click', () => {
+            if (this.currentThread) {
+                this.currentThread.redo();
+            }
+        });
+    }
+
+    /**
+     * Start a drawing stroke
+     *
+     * @param {Event} event - DOM event
+     * @return {void}
+     */
+    drawingStartHandler(event) {
```

Comment:
Starting a new drawing should trigger the creation of a blank drawing annotation and kick off the drawing path tracking in the thread